### ISSUE TITLE
check_perfmon.exe: Add fallback support for localized performance counters

### ISSUE THREAD

**Issue by dnsmichi** [Icinga/icinga2#6418]
## Expected Behavior

If the Englisch performance counter is not found, a fallback should be made available.

## Current Behavior

If a performance counter is not found, no fallback is provided. It would also help to have a more subtle error message.

## Possible Solution

Read the registry keys and do a fallback lookup ourselves. This would tremendously increase the usage of this plugin.

https://www.msxfaq.de/code/powershell/psperfcounter.htm

## Context

Localized performance counters on Windows.

**Comment by Crunsher** [comment]
This would also solve #5546

**Comment by LordHepipud** [comment]
Can we have a discussion about this please? I currently don't get your point.
As I spend the last weeks analysing the handling of performance counters and how to fetch them correctly, there might be some valuable input.

**Comment by dnsmichi** [comment]
My Windows 10 VM sources from a German license, hence it has the English language pack installed. For some reason, I cannot query performance counters in English, only German. See the URL where such a registry fallback should possibly be implemented.

The final goal is that check_perfmon.exe is capable of looking up localized performance counters, without any additional wrappers, hacks, or PS scripts.

**Comment by dnsmichi** [comment]
Ok, with the scripts provided by @LordHepipud I can see that check_perfmon.exe is doing something weird instead. The PS script works like expected on a localized system.

```
# Create a instance for our processor time, including the overall CPU usage
$counter              = New-Object System.Diagnostics.PerformanceCounter;
$counter.CategoryName = 'Processor';
$counter.CounterName  = '% Processor Time';
$counter.InstanceName = '_Total';
$counter.NextValue();
Start-Sleep -Milliseconds 500;

while ($TRUE) {
    $counter.NextValue();
    Start-Sleep -Seconds 1;
}


Einfach in PowerShell bei dir kopieren - sollte in der Regel funktionieren


# Get all available instances / counters for a specific category
$CounterCategory = 'Processor';
[hashtable]$counters = @{};
$Category = New-Object System.Diagnostics.PerformanceCounterCategory($CounterCategory);

foreach ($counter in $Category.ReadCategory().Keys) {
    [string]$CounterInstanceAddition = '';

    # As counters might also have instances (like interfaces, disks, paging file), we should
    # try to load them as well
    foreach ($instance in $Category.ReadCategory()[$counter].Keys) {
        # If we do not match this magic string, we have multiple instances we can access
        # to get informations for different disks, volumes and interfaces for example
        if ($instance -ne 'systemdiagnosticsperfcounterlibsingleinstance') {
            # Re-Write the name we return of the counter to something we can use directly
            # within our modules to load data from. A returned counter will look like this
            # for example:
            # \PhysicalDisk(*)\avg. disk bytes/read
            [string]$UsableCounterName = [string]::Format('\{0}(*)\{1}', $CounterCategory, $counter);
            if ($counters.ContainsKey($UsableCounterName) -eq $TRUE) {
                $counters[$UsableCounterName] += $Category.ReadCategory()[$counter][$instance];
            } else {
                $counters.Add($UsableCounterName, @( $Category.ReadCategory()[$counter][$instance] ));
            }
        } else {
            # For counters with no instances, we still require to return a re-build Performance Counter
            # output, to make later usage in our modules very easy. This can look like this:
            # \System\system up time
            [string]$UsableCounterName = [string]::Format('\{0}\{1}', $CounterCategory, $counter);
            $counters.Add($UsableCounterName, $null);
        }
    }
};

$counters | Out-String;
```

**Comment by dnsmichi** [comment]
Actually there's a fix available with #6800, re-opening.
